src/java.desktop/share/classes/javax/swing/JFormattedTextField.java

```diff
@@ -144,15 +144,15 @@
  * Alternatively, you could invoke <code>commitEdit</code>, which would also
  * commit the value.
  * <p>
- * <code>JFormattedTextField</code> does not do the formatting it self,
+ * <code>JFormattedTextField</code> does not do the formatting itself,
```

Comment:
Looks like we have the same typo in some other files Like SwingSet2, etc. please double-check.

Reply:
We may… I didn't scan the entire code base, I was reading the code of `JFormattedTextField` found the typo and decided to correct it. It's part of the Java SE API.

Reply:
The word _‘it self’_ is present in HTML files in SwingSet2 and SwingSet3 in the excerpts from [**Micrographia**](https://en.wikipedia.org/wiki/Micrographia). I'd rather not touch it, these are meant to be the exact text of the book, there are many misspellings or archaic spellings in the files.

Reply:
we have similar typo in the BasicTreeUI.java as well.

Reply:
> we have similar typo in the BasicTreeUI.java as well.

Yes, it's there:

https://github.com/openjdk/jdk/blob/b093880acd89d8d0bccd4b8b260b721f4dcfc161/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTreeUI.java#L4084-L4088

Yet it's not visible *externally*, it's visible only to those who read the code of the `BasicTreeUI.Handler.handleSelection` method whereas the typo I'm addressing here is in the published API documentation for [`JFormattedTextField`](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/javax/swing/JFormattedTextField.html).

It wasn't my goal to amend [**all** the instances](https://github.com/search?q=repo%3Aopenjdk%2Fjdk%20%22it%20self%22&type=code)¹ where *‘it self’* is present in the code base — it takes much more time and effort².

It was supposed to be a quick and trivial bug report and fix which addresses *this particular typo* in the single file, `JFormattedTextField.java`. A single file with ~~a single change~~ with four changed lines should've been reviewed quickly and smoothly.

For what it's worth, we have spent much more time discussing it…

¹ There are quite a few; yet they're mostly in hotspot code comments. Eventually, all could be amended.  
² I've been on this journey with [*‘a the’*](https://bugs.openjdk.org/browse/JDK-8284189) in various combinations, it took about two months since its inception to its completion.

Reply:
> For what it's worth, we have spent much more time discussing it…

Right, and I do not se a reason why other cases of the exactly same typo at least in java.desktop module is not fixed. Or do you suggest to create one more JBS issue and fix it? and then another one?

Reply:
> > For what it's worth, we have spent much more time discussing it…
> 
> Right, and I do not se a reason why other cases of the exactly same typo at least in java.desktop module is not fixed. Or do you suggest to create one more JBS issue and fix it? and then another one?

Yes, that's exactly what I suggest.

Unless you're committing to invest some *considerable* amount of time into cleaning up all the similar typos or other mistakes in the entire code base, doing a point-and-fix type of changes is a good trade-off, in my opinion, compared to ignoring them altogether. What I mean is limiting the change to *one file*, which is the smallest unit, makes it a quick fix: quick to submit a bug, quick to fix and quick to review and to integrate.

Reply:
That is wrong suggestion, if the one bug found in one place it is part of the fixing process to check that it does not exists in another one. Especially for typo it is easy to check. We already discussed this for one issue in scrollbars on windows, where the same bug was missed in the similar place.

Reply:
> That is wrong suggestion, if the one bug found in one place it is part of the fixing process to check that it does not exists in another one. Especially for typo it is easy to check.

I don't think it is wrong, especially for typos. As I said, fixing typos in the entire code base, even if limited to the `java.desktop` module requires *considerable* amount of time. Yet fixing a typo in a certain file is a *trivial fix* which is easy to fix and is quick to review.

> We already discussed this for one issue in scrollbars on windows, where the same bug was missed in the similar place.

In that case, the problem could be more serious. As I [explained](https://github.com/openjdk/jdk/pull/14338#issuecomment-1581491086) in PR #14338, I missed this function was used in another component. At the same time, no test case has been found where functionality of ScrollBar is affected.

I have submitted [the follow-up issues](https://github.com/openjdk/jdk/pull/14338#issuecomment-1706567768) to clean it up.

Reply:
Anyway, I have updated `BasicTreeUI`; in addition to fixing the typo there, I expanded the wildcard imports.

Do you have any more comments? @mrserb

Reply:
> Anyway, I have updated `BasicTreeUI`; in addition to fixing the typo there, I expanded the wildcard imports.
> 
> Do you have any more comments? @mrserb

@mrserb Have I addressed your concerns? Do you have any additional comments?